### PR TITLE
[DSIP-72][Dynamic Task] Remove dynamic task type

### LINKED ISSUE #16591: [DSIP-72][Dynamic Task] Suggest remove dynamic task type
### Search before asking

- [X] I had searched in the [DSIP](https://github.com/apache/dolphinscheduler/issues/14102) and found no similar DSIP.

### Motivation

The dynamic task type is currently supported in a user-configured manner. The user executes a while like loop to determine the exit condition. After the exit, specify multiple times of sub-workflows based on exit conditions.

The current task type has a lot of bugs, and I think the operation of executing multiple times of sub-workflows is not universal, and it is too customized for a certain business scenario. And at the same time, I search in the issue list found that only a single digit of users were using it.

So I suggest remove this dynamic task type.

### Design Detail

_No response_

### Compatibility, Deprecation, and Migration Plan

_No response_

### Test Plan

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

close #16591

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by Gallardot — DISMISSED** [review]

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/executor/plugin/dynamic/DynamicAsyncTaskExecuteFunction.java [hunk 1]
```diff
@@ -142,7 +142,6 @@ private void setOutputParameters() {
 
         List<Property> taskPropertyList = new ArrayList<>(JSONUtils.toList(taskInstance.getVarPool(), Property.class));
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/executor/plugin/dynamic/DynamicAsyncTaskExecuteFunction.java]: This directly `dynamic` can directly delete.

**Review by ruanwenjun — COMMENTED** [review]
We need to remove `WAIT_TO_RUN` state, this state is used by dynamic

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/executor/plugin/dynamic/DynamicAsyncTaskExecuteFunction.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/engine/executor/plugin/dynamic/DynamicAsyncTaskExecuteFunction.java]: Fixed.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by Gallardot — APPROVED** [review]

**Pull request merged**